ironfish-cli/src/utils/transaction.ts

```diff
@@ -36,11 +36,43 @@ export async function watchTransaction(options: {
 
   const startTime = lastTime
 
-  if (last.content.transaction == null) {
-    logger.log(`Tried to watch transaction ${options.hash} but it's missing.`)
-    return
+  let state = ''
+
+  CliUx.ux.action.start(`Current Status`)
+
+  // Loop until we find the transaction
+  while (last.content.transaction == null) {
+    state = 'not found'
+    const duration = Date.now() - lastTime
+    const span = TimeUtils.renderSpan(duration, {
+      hideMilliseconds: true,
+    })
+    CliUx.ux.action.status = `${state} ${span}`
+    logger.log(`Tried to watch transaction ${options.hash} but it's missing`)
+    logger.log(`Retrying in ${pollFrequencyMs / 1000} s...`)
+
+    await PromiseUtils.sleep(pollFrequencyMs)
+
+    last = await options.client.getAccountTransaction({
+      account: options.account,
+      hash: options.hash,
+      confirmations: options.confirmations,
+    })
+  }
```

Comment:
The logic below has an invariant that `last.content.transaction != null`. The code above could be moved within the `while (true)` loop but it would probably have to change the entire structure of the loop. 

Also, most of the code changes is UX improvements, without it the core change is ~5 lines

Reply:
We should be able to use a `prevStatus` variable that's declared outside of the loop to add this logic to the existing while loop. The loop is really just handling status transitions for the transaction, and we want to add support for a new status of "not found". 

I think the while loop needs some refactoring, but I'd expect something like this:

```
prevStatus = null
while true {
  resp = await getTransaction()
  const currentStatus = resp?.transaction ?? 'not found'

  if prevStatus != null && prevStatus != 'not found' && currentStatus == 'not found' {
    log "transaction was deleted"
    return
  }

  if currentStatus == waitUntil {
    log "reached status waitUntil"
    return
  }

  if prevStatus != null && currentStatus != prevStatus {
    log "prevStatus -> currentStatus"
  }
  
  log currentStatus xx seconds
  prevStatus = currentStatus
  sleep
}
```

Reply:
fixed the status updating